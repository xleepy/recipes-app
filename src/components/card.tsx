import { useNavigate } from 'react-router-dom';
import styles from './card.module.css';

type Props = {
  id: number;
  image: string;
  title: string;
};

export const Card = ({ id, image, title }: Props) => {
  const navigate = useNavigate();
  return (
    <div
      onClick={() => {
        navigate(`/detail/${id}`);
      }}
      className={styles.card}
    >
      <img loading="lazy" className={styles.image} src={image} />
      <p>{title}</p>
    </div>
  );
};
