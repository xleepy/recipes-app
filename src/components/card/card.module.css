.card {
  width: 100%;
  height: 100%;
  border-radius: 12px;
  overflow: auto;
  filter: drop-shadow(0 4px 4px rgb(0 0 0 / 25%));
  border: 1px solid #49454F;
  cursor: pointer;
  transition: all cubic-bezier(0.4, 0, 0.2, 1) 300ms;
}

.card:hover {
  background: #49454F;
}

.card .image {
  overflow: clip;
  aspect-ratio: 16/9;
  object-fit: cover;
  width: 100%;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
}

.card .detail {
  text-align: left;
  padding: 1rem;
}

.card .detail .title {
  margin: 0;
  font-size: 1rem;
  line-height: 24px;
  font-weight: 400;
  letter-spacing: 0.5px;
}

.card .detail p {
  font-weight: 400;
  font-size: 0.9rem;
}


.card .detail .info {
  gap: 0.5rem;
  display: flex;
}