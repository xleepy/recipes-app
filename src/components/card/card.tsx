import { useNavigate } from 'react-router-dom';
import styles from './card.module.css';

type Props = {
  id: number;
  image: string;
  title: string;
};

export const Card = ({ id, image, title }: Props) => {
  const navigate = useNavigate();
  const navigateToDetail = () => {
    navigate(`/detail/${id}`);
  };
  return (
    <div
      data-testid="card"
      onClick={navigateToDetail}
      onKeyPress={navigateToDetail}
      className={styles.card}
      role="button"
      tabIndex={0}
    >
      <img loading="lazy" alt="preview" className={styles.image} src={image} />
      <p>{title}</p>
    </div>
  );
};
