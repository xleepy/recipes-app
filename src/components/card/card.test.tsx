import { render } from '@testing-library/preact';
import { Card } from './card';
import { describe, expect, it } from 'vitest';
import { BrowserRouter } from 'react-router-dom';

describe('Card tests', () => {
  it('should render card', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Card title="test" image="" id={1} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
