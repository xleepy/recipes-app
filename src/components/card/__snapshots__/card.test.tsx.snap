// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card tests > should render card 1`] = `
<DocumentFragment>
  <div
    class="_card_c37972"
    data-testid="card"
  >
    <img
      class="_image_c37972"
      loading="lazy"
      src=""
    />
    <p>
      test
    </p>
  </div>
</DocumentFragment>
`;
